### PR TITLE
Add unit tests for tabular transformations

### PR DESCRIPTION
**Pull request by floscha**
I noticed that there are no unit tests that specifically cover the `tabular/transform.py` module. Thus, I have implemented a new `test_tabular_transform.py` test suite to cover all relevant functionality of this module, including both the `Categorify` as well as the `FillMissing` classes.

### CODE REVIEW

**Comment by sgugger** [comment]
Thanks a lot for your contribution. Could you adjust it to fit the usual fastai style? That means removing the empty lines. Then I have a few comments/suggestions of thins to add if you can that I'll put directly in the code.

File: tests/test_tabular_transform.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                 'relationship', 'race', 'sex', 'native-country']
+    df = df[cat_names]
+    train_df, valid_df = df[:1024].copy(),df[1024:1260].copy()
+    categorify_transform = Categorify(cat_names, [])
+
+    categorify_transform.apply_train(train_df)
+    categorify_transform.apply_test(valid_df)
+
+    assert all([isinstance(dt, CategoricalDtype) for dt in train_df.dtypes])
+    assert all([isinstance(dt, CategoricalDtype) for dt in valid_df.dtypes])
```
Comment (sgugger) [inline tests/test_tabular_transform.py]: Could we keep the whole df and check the columns not in cat_names aren't modified?

File: tests/test_tabular_transform.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    cont_names = ['age', 'fnlwgt', 'education-num', 'capital-gain',
+                  'capital-loss', 'hours-per-week']
+    df = df[cont_names]
+    train_df, valid_df = df[:1024].copy(),df[1024:1260].copy()
+    fill_missing_transform = FillMissing([], cont_names)
+
+    fill_missing_transform.apply_train(train_df)
+    fill_missing_transform.apply_test(valid_df)
+
+    train_df.isna().values.sum() == 0
```
Comment (sgugger) [inline tests/test_tabular_transform.py]: I think there is an assert missing there.

File: tests/test_tabular_transform.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    expected_valid_median = valid_df.median()
+    fill_missing_transform = FillMissing([], cont_names, add_col=False)
+
+    fill_missing_transform.apply_train(train_df)
+    fill_missing_transform.apply_test(valid_df)
+    train_median = train_df.median()
+    valid_median = valid_df.median()
+
+    assert np.array_equal(train_median.values, expected_train_median.values)
+    assert np.array_equal(valid_median.values, expected_valid_median.values)
```
Comment (sgugger) [inline tests/test_tabular_transform.py]: The median here could have changed since we're filling with the median of the training set, not the validation set. Maybe check that missing values are replaced by that median instead?

**Comment by floscha** [comment]
Hi @sgugger, thanks for your feedback. I agree with all your points and consequentially adapted the code. Let me know if further changes are needed. Cheers

**Comment by sgugger** [comment]
Thanks!
